### ISSUE TITLE
It turns out that stake-snapshot format has changed in node 9.1.0

### ISSUE THREAD

**Issue by matiwinnetou** [cardano-community/cncli#62]
```
{
  "pools": {
    "6c518b4861bb88b1395ceb116342cecbcfb8736282655f9a61c4c368": {
      "stakeGo": 29178182042374,
      "stakeMark": 29375050132888,
      "stakeSet": 29289980003570
    }
  },
  "total": {
    "stakeGo": 22794903139974784,
    "stakeMark": 22745376906060620,
    "stakeSet": 22810377476386930
  }
}
```

For the current epoch, this is what I do to extract actual values:

```
POOL_STAKE=$(echo $SNAPSHOT | jq -r '.pools[] | .stakeSet')
ACTIVE_STAKE=$(echo $SNAPSHOT | jq -r '.total.stakeSet')
```

**Comment by AndrewWestberg** [comment]
This has worked that way for quite a few releases I think.  You shouldn't be using JQ to parse those large numbers. It often fails to round properly.

```
STAKES=$(echo "$SNAPSHOT" | grep -oP '(?<=    "stakeSet": )\d+(?=,?)')
POOL_STAKE=$(echo $STAKES | cut -d' ' -f1)
ACTIVE_STAKE=$(echo $STAKES | cut -d' ' -f2)
```

**Comment by rphair** [comment]
Not large enough numbers in the case of Cardano's token supply, as already reported here: https://github.com/cardano-community/cncli/pull/35#discussion_r1298497770

**Comment by AndrewWestberg** [comment]
@rphair This is _not_ correct.  JQ still screws with the numbers.  JQ internally stores large numbers with IEEE floating point values which can cause this rounding issue. A future version of JQ is supposed to fix it. To demonstrate the issue, I took your values and added 1 to each of the total section.  Try these json values...

```
{
  "pools": {
    "6c518b4861bb88b1395ceb116342cecbcfb8736282655f9a61c4c368": {
      "stakeGo": 29178182042374,
      "stakeMark": 29375050132888,
      "stakeSet": 29289980003570
    }
  },
  "total": {
    "stakeGo": 22794903139974785,
    "stakeMark": 22745376906060621,
    "stakeSet": 22810377476386931
  }
}
```

Then, if you pass this file through JQ, you get...

```
$ jq . < jq.json
{
  "pools": {
    "6c518b4861bb88b1395ceb116342cecbcfb8736282655f9a61c4c368": {
      "stakeGo": 29178182042374,
      "stakeMark": 29375050132888,
      "stakeSet": 29289980003570
    }
  },
  "total": {
    "stakeGo": 22794903139974784,
    "stakeMark": 22745376906060620,
    "stakeSet": 22810377476386932
  }
}
```

The total values are all off by 1.

**Comment by rphair** [comment]
thanks @AndrewWestberg - consider me better informed now thanks to you 🙏 ... Since this is fixed in version `1.7.1` (the current release), it's still an option for operators to ensure the latest version of `jq` so the JSON can be processed semantically.
```
10:core$ jq --version
jq-1.7.1
11:core$ jq < jq.json
{
  "pools": {
    "6c518b4861bb88b1395ceb116342cecbcfb8736282655f9a61c4c368": {
      "stakeGo": 29178182042374,
      "stakeMark": 29375050132888,
      "stakeSet": 29289980003570
    }
  },
  "total": {
    "stakeGo": 22794903139974785,
    "stakeMark": 22745376906060621,
    "stakeSet": 22810377476386931
  }
}
```
`1.7.1` is currently [the release version on Ubuntu 24.04](https://launchpad.net/ubuntu/+source/jq), but it's still a tough call whether to standardise utilities on non-semantic code (`cut`), covering older distros, or require that operators install the latest `jq`.

**Comment by AndrewWestberg** [comment]
Most have not yet upgraded to 24.04 since the lts upgrade path won't be out
until October.
